src/app/project/add-edit-project.component.ts

```diff
@@ -39,14 +41,20 @@ export class AddEditProjectComponent implements OnInit {
         });
         // Else, create new project
         if (!this.edit) {
-            this.model = new ProjectForm(new Project({}));
-            // Dropdowns & autocompletes require a value, {} at the very least
-            this.model.project.scenario = {} as Scenario;
-            this.model.project.city = {} as City;
+            this.model = new ProjectForm(new Project({
+                // Dropdowns & autocompletes require a value, {} at the very least
+                'scenario': {} as Scenario,
+                'city': {} as City
+            }));
+        }
+        // Reroute to dashboard if project doesn't exist
+        if (!this.model.project) {
```

Comment:
Same thing here. Shouldn't this check be within the subscribe handler after the project is pulled?